lib/srv/desktop/tdp/proto.go

```diff
@@ -463,13 +462,48 @@ func (m Error) Encode() ([]byte, error) {
 }
 
 func decodeError(in io.Reader) (Error, error) {
-	message, err := decodeString(in, tdpMaxErrorMessageLength)
+	message, err := decodeString(in, tdpMaxNotificationMessageLength)
 	if err != nil {
 		return Error{}, trace.Wrap(err)
 	}
 	return Error{Message: message}, nil
 }
 
+type Severity byte
+
+const (
+	SeverityInfo    Severity = 0
+	SeverityWarning Severity = 1
+	SeverityError   Severity = 2
+)
+
+type Notification struct {
+	Message  string
+	Severity Severity
+}
+
+func (m Notification) Encode() ([]byte, error) {
+	buf := new(bytes.Buffer)
+	buf.WriteByte(byte(TypeNotification))
+	if err := encodeString(buf, m.Message); err != nil {
```

Comment:
Looks like the Web UI counterpart has already merged, so feel free to ignore this suggestion, but I wonder if we should stop creating new messages without a length field included ahead of time.

The pattern we've been following so far makes it nearly impossible to skip messages you don't know about and makes backwards compatibility a pain.

Reply:
I see how that could help us avoid some compatibility problems -- we can just drop messages whose opcodes we're unaware of.

I think its still a bit of a dangerous game though -- for example, if we do an update like this one, where we're now sending `tdp.Notification` rather than `tdp.Error`, then we'd run into the problem of older clients behaving weirdly because they're ignoring error messages.